### PR TITLE
closes #13034 Fixes the ScrollViewMock methods

### LINKED ISSUE #13034: Bug running Jest with ScrollView dependant components
### Description

When running `npm run test`, the default tests `index.ios.js` and `index.android.js` fail due to the following error:

```
    TypeError: _this._scrollView.scrollTo is not a function

      at TabViewPagerScroll._this._scrollTo (node_modules/react-native-tab-view/src/TabViewPagerScroll.js:68:19)
      at TabViewPagerScroll.componentDidMount (node_modules/react-native-tab-view/src/TabViewPagerScroll.js:123:197)
      at node_modules/react-test-renderer/lib/ReactCompositeComponent.js:265:25
      at measureLifeCyclePerf (node_modules/react-test-renderer/lib/ReactCompositeComponent.js:75:12)
      at node_modules/react-test-renderer/lib/ReactCompositeComponent.js:264:11
      at CallbackQueue.notifyAll (node_modules/react-test-renderer/lib/CallbackQueue.js:76:22)
      at ReactTestReconcileTransaction.close (node_modules/react-test-renderer/lib/ReactTestReconcileTransaction.js:36:26)
      at ReactTestReconcileTransaction.closeAll (node_modules/react-test-renderer/lib/Transaction.js:206:25)
      at ReactTestReconcileTransaction.perform (node_modules/react-test-renderer/lib/Transaction.js:153:16)
      at batchedMountComponentIntoNode (node_modules/react-test-renderer/lib/ReactTestMount.js:69:27)
```

### Reproduction

```sh
$ react-native init Test
$ cd Test/
$ yarn add react-navigation
$ npm run test
```

### Solution

I'm open to send a PR with a fix if somebody can guide me through this. Could be here https://github.com/facebook/react-native/blob/dbe555ba78d7b0f79e482c62787db2fc3d897848/Libraries/Components/ScrollView/__mocks__/ScrollViewMock.js?

### Additional Information

* React Native version: 0.42.0
* Platform: both
* Operating System: macOS Sierra
* Dev tools: iOS 10.2

### PR DESCRIPTION
**Pull request by alvaromb**
## Motivation (required)

Solves https://github.com/facebook/react-native/issues/13034

Now the `ScrollView` mock has all the methods available.

## Test Plan (required)

React Native tests pass.

To test this specific part of the code, 

```sh
$ react-native init Test
$ cd Test/
$ yarn add react-navigation
```

Then, add a simple project that uses `react-navigation`:

```js
import React from 'react';
import { Text } from 'react-native';
import { StackNavigator } from 'react-navigation';

class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Welcome',
  };
  render() {
    return <Text>Hello, Navigation!</Text>;
  }
}

const SimpleApp = StackNavigator({
  Home: { screen: HomeScreen },
});

export default SimpleApp
```

Run the default render tests:

```js
$ npm run test
```

### CODE REVIEW

File: Libraries/Components/ScrollView/__mocks__/ScrollViewMock.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ const requireNativeComponent = require('requireNativeComponent');
 
 const RCTScrollView = requireNativeComponent('RCTScrollView');
 
-class ScrollViewMock extends React.Component {
+const ScrollViewComponent = jest.genMockFromModule('ScrollView');
```
Comment (alvaromb) [inline Libraries/Components/ScrollView/__mocks__/ScrollViewMock.js]: I would need some guidance to avoid this `eslint` error @satya164.

Comment (satya164) [inline Libraries/Components/ScrollView/__mocks__/ScrollViewMock.js]: Add `/* eslint-env jest */` to the top

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by alvaromb** [comment]
Hey @satya164!

Can I help you with something else with this PR?

**Comment by shergin** [comment]
@alvaromb Seems this PR breaks tests. Could you please take a look?

**Comment by alvaromb** [comment]
Hi @shergin!

Sorry, I forgot about this PR. Please ping me if you need a hand with this. Thanks!

**Comment by shergin** [comment]
@alvaromb I would love to see passed tests. 😄

**Comment by alvaromb** [comment]
Hi @shergin!

This PR got merged? If not, what can I help you with to merge it?

**Comment by carlost** [comment]
@alvaromb I got bitten by this issue an burned some time trying to track it down.  a local change to how jest mocks out the `ScrollView` works but it would have been nice to have that time back :)

looks like your PR fails in CI and it now has a merge conflict with master.

## CI
i can't see the details for the failure in CircleCI ... but the build was from a few months ago so it may have been flushed.  The details for the travis-ci build failure are still available: https://travis-ci.org/facebook/react-native/jobs/255491208#L3150

The failure is:

```
node_modules/react-native/Libraries/Components/ScrollView/__mocks__/ScrollViewMock.js:23
23: const ScrollViewComponent = jest.genMockFromModule('ScrollView');
                                ^^^^ identifier `jest`. Could not resolve name
```

This occurs during the `flow check` portion of the iOS build.  Based on the date of the build i think this is the line number in the build script where the failure occurs: https://github.com/facebook/react-native/blob/29d9c35e122861d86c70b1e27a6e34eda4d82369/scripts/run-ci-e2e-tests.js#L203

So the rub is, your source file to create the mock specifies the use of flow, but flow doesn't recognize the `jest` identifier.

Two options (and I am not sure which is the react-native preference because there are examples of both in the source):
- strip off the `@flow` annotation from the file: [example of that](https://github.com/facebook/react-native/blob/master/Libraries/Utilities/__mocks__/PixelRatio.js)
- explicitly declare the `jest` identifier: [example of that](https://github.com/facebook/react-native/blob/a16ef18a803579c84c77df0274a148cafb83baaa/local-cli/util/__tests__/fs-mock-test.js#L18)

## Merge Conflict
Looks like master was updated to support a flow v0.53.  I don't think that this impacts your PR, you just need to rebase.

Let me know if you want to see this through or if you want to close out this PR and have me open up a replacement.

**Comment by alvaromb** [comment]
Just merged against master, tests fails due to other issues though.

**Comment by alvaromb** [comment]
Honestly, I don't know how I should proceed from here, should I merge against another branch? @shergin @carlost

**Comment by shergin** [comment]
¯\_(ツ)_/¯ 
Previous landing was unsuccessful because fb internal tool detected that this diff got new version on GH.

**Comment by alvaromb** [comment]
Should I open a new PR @shergin @satya164 ?

**Pull request closed without merging**